### ISSUE TITLE
Overview of installed apps and available updates

### ISSUE THREAD

**Issue by martijncalker** [argoproj/argo-cd#10618]
# Summary

I noticed that Argo CD is a brilliant platform to manage multiple Kubernetes clusters and applications at scale. It does a very good job in keeping the state of the application with the desired state in Git or Helm. 
Sometimes however, it's hard to track if all versions are up to date and figure out if there are updates available. I think it would be very helpful to provide an overview of installed applications and if there are updates available (based on sem ver in the source repositories). 

I understand that it might be out of scope for the project :) 

# Motivation

As I mentioned above, it can be tricky to keep track of all new releases. Especially in this environments (Kubernetes and Cloud Native), we're using (Helm and/or Git) repositories from all over the place (e.g.: Your Argo CD installation, your Ingress Controller, a secret engine, don't forget a service mesh, a vendor specific storage class, et cetera). They all come from different places and I noticed it's currently someone's job to manually check if there is a new version of your favourite component. 

Second: your CISO (or manager) might start asking you if you know what version of applications you are running. I believe it's a simple start for your SBOM, at Kubernetes (application) level. 

# Proposal

Argo CD is already syncing the repositories, so it should be aware of all the information. 
I'm not sure if it should be a table, batch or tooltip in the interface, an API endpoint, or use the notification service when updates arrive.

---

I would happily help with the development of a feature like this, but I have not much Golang experience and much less experience with Javascript :)

**Comment by jkleinlercher** [comment]
Hey, do you know tools like [renovate](https://github.com/renovatebot/renovate)? They can scan your gitops repo and create pull requests whenever newer versions of your apps are available.

**Comment by tomasliumparas** [comment]
Hey @martijncalker I actually had the same idea. I came up with [kupdater](https://github.com/getais/kupdater) (Kubernetes update operator) for that purpose. 
Never actually thought about seeing it in Argo. 

Renovate looks like a cool tool, but kind off difficult to configure. I wanted something more simple, just to take a glance of a running applications inside the cluster and if they have updates or not. 

It would be really cool to have that info appear in ArgoCD. For example in application list, just some sort of a badge if an update is available, not including all auto dependency patching functionality. 

I would be happy to work on that!

**Comment by martijncalker** [comment]
Yes, I think a badge would be great. I've looked into stuff like renovate as well, but it seems difficult to setup for what feels like a relative simple action. 

I just noticed that Rancher already provide this information:

![rancher-upgradable](https://docs.portworx.com/img/rancher-upgarde-app.png)
